src/utility.cpp

```diff
@@ -0,0 +1,43 @@
+#include "utils.h"
+using namespace Rcpp;
+// conversion between Rcpp data and eigen vector/matrix.
```

Comment:
Did you double check that we cannot use the `Rcpp::as` and `Rcpp::wrap` functions as mentioned here: 
https://github.com/RcppCore/RcppEigen/blob/87ef9a9f8c21ddb8a4c771c912366a155270005c/vignettes/RcppEigen-Introduction.Rnw#L163
? since we already use RcppEigen anyway

Reply:
oh I should update the wording here; `Rcpp::as` and `Rcpp::wrap` can support the conversion between `NumericVector` and `eigen::MatrixXd` but in `TMB` are actually using `TMB::matrix` (which has its base `eigen::Array`) so we still have some difficulties here using these functions.